torch/optim/lr_scheduler.py

```diff
@@ -667,18 +669,21 @@ class SequentialLR(LRScheduler):
     """
 
     def __init__(self, optimizer, schedulers, milestones, last_epoch=-1, verbose="deprecated"):
-        for scheduler_idx in range(len(schedulers)):
-            if schedulers[scheduler_idx].optimizer != optimizer:
+        if len(schedulers) < 1:
+            raise ValueError("Sequential Schedulers expects at least one scheduler, but got no scheduler.")
+
+        for scheduler_idx, scheduler in enumerate(schedulers):
+            if not isinstance(scheduler, LRScheduler):
+                raise TypeError(
+                    f"{self.__class__.__name__} expects all schedulers to be of type LRScheduler, "
+                    f"but got {type(scheduler)}"
+                )
+            if scheduler.optimizer != optimizer:
```

Comment:
This is also BC breaking. Previously, no optimizer really needs to be passed in (it could be kept as None), but now the optimizer needs to be passed into SequentialLR. We'd probably want to relax this check to allow for more flexibility

Reply:
Hi @janeyx99, I think the logic here is still the same as the original [L671](https://github.com/pytorch/pytorch/blob/main/torch/optim/lr_scheduler.py#L671). I do agree on unify the logic between `SequentialLR` and `ChainedScheduler` 's `__init__`, to set the input `optimizer` as optional, however, this would probably break others code bases.